src/psyclone/psyir/nodes/acc_directives.py

```diff
@@ -518,18 +518,26 @@ def validate_global_constraints(self):
         Perform validation of those global constraints that can only be done
         at code-generation time.
 
-        :raises GenerationError: if this ACCLoopDirective is not enclosed \
-                            within some OpenACC parallel or kernels region.
+        :raises GenerationError: if this ACCLoopDirective is not enclosed
+            within some OpenACC parallel or kernels region and is not in a
+            Routine that has been marked up with an 'ACC Routine' directive.
         '''
         # It is only at the point of code generation that we can check for
         # correctness (given that we don't mandate the order that a user can
         # apply transformations to the code). As an orphaned loop directive,
         # we must have an ACCParallelDirective or an ACCKernelsDirective as
         # an ancestor somewhere back up the tree.
```

Comment:
The ACCRoutine case is missing from this comment, but tbh I would just remove all this comment as this is already explained in the docstring and the code is quite clear.

Reply:
'tis gone.